src/actions/Bluetooth/SensorUpdateActions.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2021
+ */
+
+import { PermissionActions } from '../PermissionActions';
+import BleService from '../../bluetooth/BleService';
+
+export const UPDATE_ACTIONS = {
+  SET_LOG_INTERVAL_ERROR: 'Bluetooth/setLogIntervalError',
+  SET_LOG_INTERVAL_START: 'Bluetooth/setLogIntervalStart',
+  SET_LOG_INTERVAL_SUCCESS: 'Bluetooth/setLogIntervalSuccess',
+  DISABLE_BUTTON_START: 'Bluetooth/disableButtonStart',
+  DISABLE_BUTTON_STOP: 'Bluetooth/disableButtonStop',
+};
+
+const setLogIntervalStart = macAddress => ({
+  type: UPDATE_ACTIONS.SET_LOG_INTERVAL_START,
+  payload: { macAddress },
+});
+const setLogIntervalSuccess = () => ({ type: UPDATE_ACTIONS.SET_LOG_INTERVAL_SUCCESS });
+
+const setLogIntervalError = () => ({ type: UPDATE_ACTIONS.SET_LOG_INTERVAL_ERROR });
+
+const disableButtonStart = macAddress => ({
+  type: UPDATE_ACTIONS.DISABLE_BUTTON_START,
+  payload: { macAddress },
+});
+
+const disableButtonStop = macAddress => ({
+  type: UPDATE_ACTIONS.DISABLE_BUTTON_STOP,
+  payload: { macAddress },
+});
+
+const setLogInterval = (macAddress, interval) => async dispatch => {
+  dispatch(setLogIntervalStart(macAddress));
+
+  try {
+    const regex = new RegExp(`Interval: ${interval}s`); // TODO: update with sensor specific response as needed
+    const error = `Sensor response was not equal to 'Interval: ${interval}s'`;
+    const response = await BleService().updateLogIntervalWithRetries(macAddress, interval, 10);
+    const action = regex.test(response.toString())
+      ? setLogIntervalSuccess()
+      : setLogIntervalError(error);
```

Comment:
I just noticed this and it doesn't even matter as we don't seem to be using the payload, but it's a little odd to have `error` sometimes be a string and sometimes be an error object